File: app/src/main/java/me/kashyap/theverge/BaseActivity.java
package me.kashyap.theverge;

import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;

/**
 * Created on 3/26/2015.
 */
public abstract class BaseActivity extends ActionBarActivity {

    private UiComponent uiComponent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        uiComponent = Dagger_UiComponent.builder()
                .apiModule(new ApiModule())
                .vergeModule(new VergeModule())
                .uiModule(new UiModule()).build();
    }

    UiComponent getUiComponent() {
        return uiComponent;
    }
}

File: app/src/main/kotlin/me/kashyap/theverge/BaseActivity.kt
package me.kashyap.theverge

import android.os.Bundle
import android.support.v7.app.ActionBarActivity
import android.util.Log

/**
 * Created on 3/26/2015.
 */
public abstract class BaseActivity : ActionBarActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("Base Activity", "onCreate called")
//        RssApplication.getApp(this).appComponent.inject(this)
    }
}